NFluent.Json.Tests/JsonElementGuidPropertyCheckShould.cs

```diff
@@ -0,0 +1,94 @@
+using NFluent.Helpers;
+using Xunit;
+
+namespace NFluent.Json.Tests;
+
+public class JsonElementGuidPropertyCheckShould
+{
+    [Fact]
+    public async Task HasGuidPropertyWorksOnHavingPropertyWithExpectedValue()
+    {
+        var expectedValue = Guid.NewGuid();
+        var json = await TestJson.Element(new { propA = expectedValue });
+
+        Check.That(json).HasGuidProperty("propA", expectedValue);
+    }
+
+    [Fact]
+    public async Task HasGuidPropertyFailingWhenPropertyIsUndefined()
+    {
+        var json = await TestJson.Element(new { propA = "" });
+
+        Check.ThatCode(() => Check.That(json).HasGuidProperty("propB", Guid.NewGuid())).IsAFailingCheckWithMessage(
+            "",
+            "The 'propB' property is undefined.",
+            "The checked struct:",
+            "\t[{\"propA\":\"\"}]");
+    }
+
+    [Fact]
+    public async Task HasGuidPropertyFailingWhenPropertyIsNotAGuid()
+    {
+        var json = await TestJson.Element(new { propA = "42" });
+        var expectedValue = Guid.NewGuid();
+
+        Check.ThatCode(() => Check.That(json).HasGuidProperty("propA", expectedValue)).IsAFailingCheckWithMessage(
+            "",
+            "The 'propA' property kind is not string.",
```

Comment:
The message does not match what is actually tested, I would use something like  `The 'propA' property is not a guid`.